tests/passw_hardening/test_passw_hardening.py

```diff
@@ -0,0 +1,682 @@
+                        len_min='8', reject_user_passw_match='true', lower_class='true',
+                        upper_class='true', digit_class='true', special_class='true'):
+
+        self.policies = {
+                       "state": state,
+                       "expiration": expiration,
+                       "expiration-warning": expiration_warning,
+                       "history-cnt": history,
+                       "len-min": len_min,
+                       "reject-user-passw-match": reject_user_passw_match,
+                       "lower-class": lower_class,
+                       "upper-class": upper_class,
+                       "digits-class": digit_class,
+                       "special-class": special_class
+        }
+
+
+def run_diff(file1, file2):
+    try:
+        diff_out = subprocess.check_output('diff -ur {} {} || true'.format(file1, file2), shell=True)
+        return diff_out
+    except subprocess.CalledProcessError as err:
+        syslog.syslog(syslog.LOG_ERR, "{} - failed: return code - {}, output:\n{}".format(err.cmd, err.returncode, err.output))
+        return -1
+
+def run_remote_command(duthost, command):
+    '''Running cmd from remote (switch).
+    If the command fail, the test module will not crash, the function will return false, other true'''
+    cmd = None
+    try:
+        logging.debug('DUT command = {}'.format(command))
+        cmd = duthost.shell(command)
+        logging.debug('command output lines = {}'.format(cmd["stdout_lines"]))
+        return True, cmd["stdout_lines"]
+    except Exception as e:
+        logging.error('command={} fail, error={}'.format(command, e))
+        return False, e.results['stderr_lines']
+
+def config_user(duthost, username, password=None, mode='add'):
+    username = username.strip()
+    cmd_res = False, False
+    if mode == 'add':
+        command = "useradd {}".format(username)
+        res_add_user = run_remote_command(duthost, command)
+        # TODO: add some condition about user succcess
+        cmd_res = change_username_password(duthost, password, username)
+    elif mode == 'del':
+        command = "userdel {}".format(username)
+        cmd_res = run_remote_command(duthost, command)
+    return cmd_res
+
+def change_username_password(duthost, password, username):
+    passwd_cmd = "echo -e \'"+password+"\' | passwd "+username
+    cmd_res = run_remote_command(duthost, passwd_cmd)
+    return cmd_res
+
+def get_user_expire_time_global(duthost, age_type):
+    """ Function verify that the current age expiry policy values are equal from the old one
+        Return update_age_status 'True' value meaning that was a modification from the last time, and vice versa.
+    """
+    days_num = -1
+    regex_days = AGE_DICT[age_type]['REGEX_DAYS']
+    days_type = AGE_DICT[age_type]['DAYS']
+
+    # get login.def from remote switch
+    command = regex_days+ ' /etc/login.defs'
+    try:
+        grep_max_days = duthost.command(command)
```

Comment:
grep_max_days_out  = duthost.command(command)["stdout_lines"]

and use:  grep_max_days_out  instead  grep_max_days["stdout_lines"]